Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -228,6 +240,40 @@ struct Gx2FitterResult {
   const TrackingVolume* startVolume = nullptr;
 };
 
+/// @brief A container to store scattering properties for each material surface
+///
+/// This struct holds the scattering angles, the inverse covariance of the
+/// material, and a validity flag indicating whether the material is valid for
+/// the scattering process.
+struct ScatteringProperties {
+  /// @brief Constructor to initialize scattering properties.
+  ///
+  /// @param scatteringAngles_ The vector of scattering angles.
+  /// @param invCovarianceMaterial_ The inverse covariance of the material.
+  /// @param materialIsValid_ A boolean flag indicating whether the material is valid.
+  ScatteringProperties(const BoundVector& scatteringAngles_,
+                       const ActsScalar invCovarianceMaterial_,
+                       const bool materialIsValid_)
+      : scatteringAngles(scatteringAngles_),
+        invCovarianceMaterial(invCovarianceMaterial_),
+        materialIsValid(materialIsValid_) {}
+
+  /// No default construction allowed.
+  ScatteringProperties() = delete;
+
+  /// Vector of scattering angles. The vector is usually all zeros except for
+  /// eBoundPhi and eBoundTheta.
+  BoundVector scatteringAngles;
+
+  /// Inverse covariance of the material. Compute with e.g. the Highland
+  /// formula.
+  const ActsScalar invCovarianceMaterial;
+
+  /// Flag indicating whether the material is valid. Commonly vacuum and zero
+  /// thickness material will be ignored.
+  const bool materialIsValid;
```

Comment:
Is `const` strictly necessary? This will prohibit default move. Alternatives are
1. just leave it mutable
2. create accessor functions

Reply:
Option 2, because I feel safer, having them unmutable.